sdk/typescript/src/builder/TransactionBlock.ts

```diff
@@ -624,16 +624,22 @@ export class TransactionBlock {
           provider,
         );
       }
-
       if (!this.blockData.gasConfig.budget) {
         const dryRunResult = await expectProvider(
           provider,
         ).dryRunTransactionBlock({
           transactionBlock: this.#blockData.build({
-            overrides: { gasConfig: { budget: String(MAX_GAS), payment: [] } },
+            overrides: {
+              gasConfig: {
+                budget: String(MAX_GAS),
+                // use user provided payment if it exists.
+                payment: this.#blockData.gasConfig.payment
```

Comment:
FWIW this was intentionally empty to guarantee this succeeds regardless of the payment that's attached to it or not. I don't have a strong preference for if this should be the case or not, but that was the original idea behind it.